### PR TITLE
Add "stock-check" command

### LINKED ISSUE #92: Add Restock Reminder
Title.

### PR DESCRIPTION
**Pull request by yijiano**
Closes #92 and #107

- "stock-check" command implemented. Queries the list for items that are below a certain quantity.
- Updated help command output
- Added restock reminder on startup
- fixed the following bug

image

### CODE REVIEW

File: src/main/java/seedu/pill/util/Printer.java [hunk 1]
```diff
@@ -28,9 +28,17 @@ public static void printSpace(){
 
     /**
      * Initializes the bot, prints the logo and a welcome message.
+     *
+     * @param items Reference ItemMap to print restock list.
+     * @param threshold The minimum number of items before it is deemed to require replenishment.
      */
```
Comment (cnivedit) [inline src/main/java/seedu/pill/util/Printer.java:34]: Should the javadoc be updated to reflect that the method prints items to be restocked?

**Review by cnivedit — APPROVED** [review]
Few nits to fix

File: src/main/java/seedu/pill/command/HelpCommand.java [hunk 2]
```diff
@@ -15,9 +15,11 @@
...
 public class HelpCommand extends Command {
     private static final Logger logger = PillLogger.getLogger();
-    private static final List<String> VALID_COMMANDS = Arrays.asList("help", "add", "delete", "edit", "list", "exit");
-    private String commandName;
-    private boolean verbose;
+    private static final List<String> VALID_COMMANDS = Arrays.asList(
+            "help", "add", "delete", "edit",
+            "list", "exit", "stock-check");
+    private final String commandName;
+    private final boolean verbose;
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/command/HelpCommand.java:22]: Good job making the indentation of this follow coding standards

File: src/main/java/seedu/pill/command/StockCheckCommand.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import seedu.pill.util.ItemMap;
+import seedu.pill.util.Storage;
+
+public class StockCheckCommand extends Command{
+    private final int threshold;
+
+    public StockCheckCommand(String threshold) {
+        this.threshold = Integer.parseInt(threshold);
+    }
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/command/StockCheckCommand.java:11]: Should this catch an exception? If I recall correctly, Integer.parseInt throws an exception if threshold is not valid.

File: src/main/java/seedu/pill/util/Parser.java [hunk 4]
```diff
@@ -63,8 +64,17 @@ public void parseCommand(String input) {
                 new HelpCommand(argument, flag).execute(this.items, this.storage);
                 break;
             case "list":
+                if (splitInput.length > 1) {
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
```
Comment (yakultbottle) [inline src/main/java/seedu/pill/util/Parser.java:69]: Good job adding more guard clauses, but could this be abstracted into another method?

**Review by yakultbottle — APPROVED** [review]
LGTM, only a few small nits

**Pull request merged**